internal_ws/yktrace/src/tir.rs

```diff
@@ -153,10 +154,30 @@ impl<'a, 'm> TirTrace<'a, 'm> {
                             idx: rnm.rename_irplace(idx, &body),
                             scale: *scale,
                         },
-                        Statement::Store(dst, src) => Statement::Store(
-                            rnm.rename_irplace(dst, &body),
-                            rnm.rename_irplace(src, &body),
-                        ),
+                        Statement::Store(dst, src) => {
+                            if matches!(dst, IRPlace::Val { local, .. } if local == &sir::RETURN_LOCAL)
+                            {
+                                if body.flags.contains(BodyFlags::INTERP_STEP) {
+                                    match src {
+                                        IRPlace::Const { val: Constant::Bool(b), ..} => {
+                                            // If the `interp_step` function returns false, we
+                                            // enabled trace looping which loops the trace
+                                            // indefinitely until a guard fails.
+                                            if *b == false {
+                                                loop_trace = true;
+                                            }
+                                        }
+                                        _ => {
+                                            panic!("Return value of `interp_step` function must be constant.")
```

Comment:
We still need to revise this comment. See [here](https://github.com/softdevteam/yk/pull/289#commitcomment-48317368).

Reply:
Fixed in 388b3c3.